src/charm.py

```diff
@@ -250,7 +250,8 @@ def _update_workload_container_config(self, event):
         # This token comes from an action rather than config so we check for it specifically.
         if not self.config.get("server.is-hosted"):
             is_airgapped = self._get_available_pro_airgapped_server_address() is not None
-            if not is_airgapped and not self._state.resource_token:
+            # blocked status if not airgapped, no resource token in the state and no sync token in the config
+            if not is_airgapped and not self._state.resource_token and not self.config.get("patch-sync.token"):
```

Comment:
Not the best charm code ever. I.e. Would be nice to put these blocks of logic somewhere more readable. But this is okay for now.

Reply:
do we have a test for this?

Reply:
there was the negative test (missing token -> blocked state), the test that was missing was (sync token -> active state).
I've added it